Sources/StreamChat/Workers/UserUpdater.swift

```diff
@@ -27,6 +27,62 @@ class UserUpdater: Worker {
             completion?($0.error)
         }
     }
+    
+    /// Blocks the user with the provided `userId`.
+    /// - Parameters:
+    ///   - userId: The user identifier.
+    ///   - keepChannelsVisible: A Boolean value that  determines if the channels should be kept visible.
+    ///   - completion: Called when the API call is finished. Called with `Error` if the remote update fails.
+    ///
+    func blockUser(_ userId: UserId, keepChannelsVisible: Bool = false, completion: ((Error?) -> Void)? = nil) {
+        apiClient.request(endpoint: .blockUser(userId, keepChannelsVisible: keepChannelsVisible)) {
+            switch $0 {
+            case let .success(payload):
+                self.database.write({ session in
+                    var blockedUsers: Set<BlockedUserDTO> = []
+                    for blockedUser in payload.blockedUsers {
+                        blockedUsers.insert(blockedUser.asDTO(context: self.database.writableContext))
+                    }
+                    session.currentUser?.blockedUsers = blockedUsers
+                    session.currentUser?.user.blockedUserIds = blockedUsers.map { $0.blockedUserId }
+                }, completion: {
+                    if let error = $0 {
+                        log.error("Failed to save blocked user with id: <\(userId)> to the database. Error: \(error)")
+                    }
```

Comment:
Ideally if the request fails, we should remove the user from the blocked users

Reply:
I added this logic, but I'm not sure I've chosen the best approach.